src/jit/lsraxarch.cpp

```diff
@@ -1341,123 +1329,111 @@ int LinearScan::BuildBlockStore(GenTreeBlk* blkNode)
     }
     else
     {
-        // CopyObj or CopyBlk
-        if (source->gtOper == GT_IND)
+        if (src->OperIs(GT_IND))
         {
-            assert(source->isContained());
-            srcAddrOrFill = source->gtGetOp1();
+            assert(src->isContained());
+            srcAddrOrFill = src->AsIndir()->Addr();
         }
-        if (blkNode->OperGet() == GT_STORE_OBJ)
+
+        if (blkNode->OperIs(GT_STORE_OBJ))
         {
             if (blkNode->gtBlkOpKind == GenTreeBlk::BlkOpKindRepInstr)
             {
                 // We need the size of the contiguous Non-GC-region to be in RCX to call rep movsq.
-                blkSizeRegMask = RBM_RCX;
+                sizeRegMask = RBM_RCX;
             }
+
             // The srcAddr must be in a register.  If it was under a GT_IND, we need to subsume all of its
             // sources.
-            sourceRegMask  = RBM_RSI;
             dstAddrRegMask = RBM_RDI;
+            srcRegMask     = RBM_RSI;
         }
         else
         {
             switch (blkNode->gtBlkOpKind)
             {
                 case GenTreeBlk::BlkOpKindUnroll:
-                    // If we have a remainder smaller than XMM_REGSIZE_BYTES, we need an integer temp reg.
-                    //
-                    // x86 specific note: if the size is odd, the last copy operation would be of size 1 byte.
-                    // But on x86 only RBM_BYTE_REGS could be used as byte registers.  Therefore, exclude
-                    // RBM_NON_BYTE_REGS from internal candidates.
-                    if ((size & (XMM_REGSIZE_BYTES - 1)) != 0)
+                    if ((size % XMM_REGSIZE_BYTES) != 0)
                     {
                         regMaskTP regMask = allRegs(TYP_INT);
-
 #ifdef _TARGET_X86_
                         if ((size & 1) != 0)
                         {
-                            regMask &= ~RBM_NON_BYTE_REGS;
+                            // We'll need to store a byte so a byte register is needed on x86.
+                            regMask = allByteRegs();
                         }
 #endif
                         buildInternalIntRegisterDefForNode(blkNode, regMask);
                     }
 
                     if (size >= XMM_REGSIZE_BYTES)
                     {
-                        // If we have a buffer larger than XMM_REGSIZE_BYTES,
```

Comment:
Were these comments wrong or unnecessary?

Reply:
The commenting throughout the JIT codebase is in general pretty terrible. The first rule about commenting is that comments should explain why the code does something when that isn't obvious and NOT what the code does. What the code does should be obvious from, well, the code itself.

Now, in complex code it's not always possible to write clear enough code. Or sometimes the code is clear but is large and saying a few words about what it does is helpful because it speeds up understanding of the code.

That's not the case here. It's 3 lines of trivial code and 5 lines of comments.